test/metabase/query_processor_test.clj

```diff
@@ -45,7 +45,8 @@
   (let [features (set (cons feature more-features))]
     (set (for [driver (normal-drivers)
                :let   [driver (tx/the-driver-with-test-extensions driver)]
-               :when  (set/subset? features (driver.u/features driver (data/db)))]
+               :when  (driver/with-driver driver
+                        (set/subset? features (driver.u/features driver (data/db))))]
```

Comment:
I would consider doing it like this instead. `driver.u/features` calls `driver/database-supports?` for every single feature that exists (currently 28 but bound to grow in the future), and almost every time we use this we only need to know about one or two features. Hopefully `database-supports?` methods aren't doing anything too expensive but just in case any of them are, this could make a big performance difference since we're doing it over and over again in hundreds of tests

```suggestion
               :let   [db (data/db)]
               :when  (driver/with-driver driver
                        (every? #(driver/database-supports? driver % db) features))]
```

I tried it with just Postgres and H2 locally and doing it about a thousand times is about a 150 ms faster, but my local computer is a lot faster than the CI ones. That's not much but in a few years if we have 100 features and twice as 2000 tests using this tool we're talking about a difference of a few seconds when running the full test suite